### PR TITLE
2019 Nissan Altima fingerprint

### PR DESCRIPTION
**Pull request by Hana2736**
Added 2019 Altima fingerprint and updated readme. Not sure how to denote that this is the turbocharged model, so there may be some differences in fingerprint between that and NA.

Tested with f4ff336f2ad085ce|2021-05-20--20-30-47

### CODE REVIEW

File: selfdrive/car/nissan/values.py [hunk 1]
```diff
@@ -52,6 +52,10 @@ class CAR:
     {
       2: 5, 42: 6, 346: 6, 347: 5, 348: 8, 349: 7, 386: 8, 397: 8, 398: 8, 520: 2, 523: 6, 548: 8, 634: 7, 645: 8, 658: 8, 665: 8, 666: 8, 674: 2, 682: 8, 683: 8, 689: 8, 723: 8, 758: 3, 772: 8, 773: 6, 774: 7, 775: 8, 776: 6, 777: 7, 778: 6, 783: 3, 851: 8, 855: 5, 1001: 6, 1041: 8, 1042: 8, 1055: 3, 1100: 7, 1104: 4, 1105: 6, 1107: 4, 1108: 8, 1110: 7, 1111: 7, 1227: 8, 1228: 8, 1229: 8, 1232: 8, 1247: 4, 1266: 8, 1273: 7, 1306: 1, 1342: 1, 1376: 8, 1401: 8, 1497: 3, 1514: 6, 1526: 8, 1527: 5, 1792: 8, 1821: 8, 1823: 8, 1837: 8, 1872: 8, 1937: 8, 1953: 8, 1968: 8, 1988: 8, 2000: 8, 2001: 8, 2004: 8, 2005: 8, 2015: 8, 2016: 8, 2017: 8, 2024: 8, 2025: 8
     },
+    #2019 Altima VC-Turbo
+    {
```
Comment (adeebshihadeh) [inline selfdrive/car/nissan/values.py:56]: merge your fingerprint with the existing one

File: selfdrive/car/nissan/values.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ class CAR:
...
..., 346: 6, 347: 5, 348: 8, 349: 7, 386: 8, 397: 8, 398: 8, 520: 2, 523: 6, 548: 8, 634: 7, 645: 8, 658: 8, 665: 8, 666: 8, 674: 2, 682: 8, 683: 8, 689: 8, 723: 8, 758: 3, 772: 8, 773: 6, 774: 7, 775: 8, 776: 6, 777: 7, 778: 6, 783: 3, 851: 8, 855: 5, 1001: 6, 1041: 8, 1042: 8, 1055: 3, 1100: 7, 1104: 4, 1105: 6, 1107: 4, 1108: 8, 1110: 7, 1111: 7, 1227: 8, 1228: 8, 1229: 8, 1232: 8, 1247: 4, 1266: 8, 1273: 7, 1306: 1, 1342: 1, 1376: 8, 1401: 8, 1497: 3, 1514: 6, 1526: 8, 1527: 5, 1792: 8, 1821: 8, 1823: 8, 1837: 8, 1872: 8, 1937: 8, 1953: 8, 1968: 8, 1988: 8, 2000: 8, 2001: 8, 2004: 8, 2005: 8, 2015: 8, 2016: 8, 2017: 8, 2024: 8, 2025: 8, 397: 8, 398: 8, 674: 2, 1376: 8, 1497: 3, 2025: 8, 2024: 8, 758: 3, 1937: 8, 1110: 7, 1792: 8, 2001: 8, 2000: 8, 1273: 7, 682: 8, 1107: 4, 1228: 8, 1227: 8, 1821: 8, 520: 2, 523: 6, 1266: 8, 645: 8, 689: 8, 1100: 7, 1306: 1, 1823: 8, 1988: 8, 1042: 8, 1041: 8, 1953: 8, 772: 8, 773: 6, 774: 7, 851: 8, 2005: 8, 775: 8, 1872: 8, 2004: 8, 776: 6, 777: 7, 1111: 7, 1232: 8, 778: 6, 855: 5, 1837: 8, 1055: 3, 386: 8, 783: 3, 1247: 4, 1401: 8, 2017: 8, 2016: 8, 346: 6, 665: 8, 2015: 8, 347: 5, 666: 8, 348: 8, 349: 7, 548: 8, 1968: 8, 42: 6, 1527: 5, 1526: 8
```
Comment (adeebshihadeh) [inline selfdrive/car/nissan/values.py:53]: there's duplicate keys now

File: selfdrive/car/nissan/values.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ class CAR:
...
...4: 2, 682: 8, 683: 8, 689: 8, 723: 8, 758: 3, 772: 8, 773: 6, 774: 7, 775: 8, 776: 6, 777: 7, 778: 6, 783: 3, 851: 8, 855: 5, 1001: 6, 1041: 8, 1042: 8, 1055: 3, 1100: 7, 1104: 4, 1105: 6, 1107: 4, 1108: 8, 1110: 7, 1111: 7, 1227: 8, 1228: 8, 1229: 8, 1232: 8, 1247: 4, 1266: 8, 1273: 7, 1306: 1, 1342: 1, 1376: 8, 1401: 8, 1497: 3, 1514: 6, 1526: 8, 1527: 5, 1792: 8, 1821: 8, 1823: 8, 1837: 8, 1872: 8, 1937: 8, 1953: 8, 1968: 8, 1988: 8, 2000: 8, 2001: 8, 2004: 8, 2005: 8, 2015: 8, 2016: 8, 2017: 8, 2024: 8, 2025: 8, 709: 7, 590: 8, 591: 8, 592: 8, 750: 8, 1259: 8, 1258: 8, 675: 8, 676: 8, 1454: 8, 711: 7, 712: 7, 438: 8, 713: 7, 714: 8, 517: 8, 638: 8, 715: 8, 716: 8, 717: 7, 718: 7, 719: 7, 361: 8, 684: 8, 685: 8, 686: 8, 522: 8, 687: 8, 720: 7, 600: 8, 1145: 7, 403: 8, 601: 8, 1144: 7, 648: 5, 726: 7, 727: 7, 728: 7, 690: 8, 570: 8, 451: 8, 1314: 8, 576: 8, 577: 8, 610: 8, 654: 6, 611: 8, 612: 8, 614: 8, 735: 8, 779: 7, 615: 8, 659: 8, 539: 8, 616: 8, 617: 8, 582: 8, 583: 8, 660: 8, 781: 7, 584: 8, 661: 8, 782: 7, 541: 7, 542: 8, 586: 8, 1324: 8, 389: 8, 543: 8, 587: 8, 1323: 8, 544: 8, 588: 8, 545: 8, 589: 8, 622: 8, 546: 8, 623: 8, 547: 8, 746: 8, 703: 8, 748: 6, 749: 6, 708: 7
```
Comment (pd0wm) [inline selfdrive/car/nissan/values.py:53]: Can you sort the keys?

**Comment by adeebshihadeh** [comment]
Can you plug your device in and make sure the "Upload Raw Logs" toggle is enabled? There are a lot of new values, and I'd like to make sure they're all necessary.

**Comment by EwanMac3** [comment]
> Can you plug your device in and make sure the "Upload Raw Logs" toggle is enabled? There are a lot of new values, and I'd like to make sure they're all necessary.

It should have everything uploading, whenever I drive it's connected online and I have the logging setting on.

**Pull request merged**